src/graphql/schemas/args/argGenerator.ts

```diff
@@ -0,0 +1,177 @@
+import { ClassType, Field, InputType } from "type-graphql";
+import { SortOrder } from "../enums/sortEnums.js";
+import {
+  BigIntSearchOptions,
+  BooleanSearchOptions,
+  IdSearchOptions,
+  NumberArraySearchOptions,
+  NumberSearchOptions,
+  StringArraySearchOptions,
+  StringSearchOptions,
+} from "../inputs/searchOptions.js";
+
+type ReferenceDefinition = {
+  type: keyof SearchOptionType;
+  references: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    entity: ClassType<any>;
+    fields?: Record<string, keyof SearchOptionType>;
+  };
+};
+
+export type SearchOptionType = {
```

Comment:
You can derive this from `SearchOptionMap`

```
type Bla = { [K in keyof typeof SearchOptionMap]: typeof SearchOptionMap[K] }
```

Reply:
Good catch. But actually removed it as it was no longer used 🙈 

Change in the next push